modules/artifact/src/test/java/org/mule/runtime/module/artifact/classloader/IBMMQResourceReleaserTestCase.java

```diff
@@ -333,4 +518,51 @@ private int countJULKnownLevels(ClassLoader artifactClassLoader) throws Exceptio
     return counter;
   }
 
+  private int countWorkerThreads(ClassLoader artifactClassLoader) {
+    ThreadUtils.getAllThreads().stream().forEach(o -> {
+      System.out.println(o.getName());
+    });
+
+    List<Thread> threads = ThreadUtils.getAllThreads().stream()
+        .filter(thread -> thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME))
+        .filter(thread -> thread.getClass().getClassLoader() == artifactClassLoader)
+        .collect(Collectors.toList());
+    return threads.size();
+  }
+
+  private void createWorkerThread(MuleArtifactClassLoader artifactClassLoader) throws ClassNotFoundException {
+    // Load all the clases need to create the IBM Worker thread.
+    artifactClassLoader.loadClass(IBM_WORKER_CLASS);
+    artifactClassLoader.loadClass("com.ibm.msg.client.commonservices.workqueue.WorkQueueItem");
+    artifactClassLoader.loadClass("com.ibm.msg.client.commonservices.workqueue.SimpleWorkQueueItem");
+    artifactClassLoader.loadClass("com.ibm.msg.client.commonservices.j2se.workqueue.WorkQueueManagerImplementation");
+    artifactClassLoader
+        .loadClass("com.ibm.msg.client.commonservices.j2se.workqueue.WorkQueueManagerImplementation$WorkQueueManagerThread");
```

Comment:
All these loads seem unnecessary. Shouldn't it be enough with the Class.forName below?

Reply:
Actually they are needed  I was adding these ones one by one to be sure they are needed the WorkQueueManagerImplementation needs to be loaded to the thread to run and the WorkQueueManagerImplementation need them